compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -302,6 +341,18 @@ def AMDAIE_NpuDmaCpyNdOp: AMDAIE_Op<"npu.dma_cpy_nd",
       return memSpace ? cast<IntegerAttr>(memSpace).getInt() : 0;
     }
 
+    BdIdOp getSourceBdIdOp() {
+      Value bdIdValue = getSourceBdId();
+      if (!bdIdValue || !bdIdValue.getDefiningOp()) return nullptr;
+      return dyn_cast<BdIdOp>(bdIdValue.getDefiningOp());
```

Comment:
is the reason to not make this a `cast` (and instead catch `null` outside) in order to emit a legible error message?

Reply:
Yeah